### ISSUE TITLE
GU0014 should warn here

### ISSUE THREAD

**Issue by JohanLarsson** [GuOrg/Gu.Analyzers#219]
```cs
namespace RoslynSandbox
{
    public class C
    {
        private readonly string s1;
        private readonly string s2;

        public C(string s1, string s2)
        {
            this.s1 = s1;
            this.s2 = s2;
            var a = this.s1; // prefer parameter
        }
    }
}
```

**Comment by JohanLarsson** [comment]
```cs
        [Test]
        public static void UsingFieldAfterAssigningOther()
        {
            var before = @"
namespace N
{
    public class C
    {
        private readonly string s1;
        private readonly string s2;

        public C(string s1, string s2)
        {
            this.s1 = s1;
            this.s2 = s2;
            var a = ↓this.s1;
        }
    }
}";

            var after = @"
namespace N
{
    public class C
    {
        private readonly string s1;
        private readonly string s2;

        public C(string s1, string s2)
        {
            this.s1 = s1;
            this.s2 = s2;
            var a = s1;
        }
    }
}";
            RoslynAssert.CodeFix(Analyzer, Fix, ExpectedDiagnostic, before, after);
        }
```
